adapters/repos/db/migrator_shard_status_ops.go

```diff
@@ -0,0 +1,217 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	command "github.com/weaviate/weaviate/cluster/proto/api"
+	"github.com/weaviate/weaviate/cluster/types"
+	"github.com/weaviate/weaviate/entities/errorcompounder"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+	"github.com/weaviate/weaviate/entities/models"
+)
+
+func (m *Migrator) frozen(idx *Index, frozen []string, ec *errorcompounder.ErrorCompounder) {
+	if m.cluster == nil {
+		ec.Add(fmt.Errorf("no cluster exists in the migrator"))
+		return
+	}
+
+	idx.backupMutex.RLock()
```

Comment:
Should we rename the mutex?

Reply:
i agree, good catch

Reply:
if the diff not much, will make it in a follow up PR, because my original idea was to minmize the diff for reviewing. wdyt ?

Reply:
https://github.com/weaviate/weaviate/pull/5144